library/azure_rm_virtualmachine.py

```diff
@@ -1341,18 +1364,18 @@ def delete_vm(self, vm):
             self.log('Storing NIC names for deletion.')
             for interface in vm.network_profile.network_interfaces:
                 id_dict = azure_id_to_dict(interface.id)
-                nic_names.append(id_dict['networkInterfaces'])
-            self.log('NIC names to delete {0}'.format(', '.join(nic_names)))
+                nic_names.append(dict(name=id_dict['networkInterfaces'], resource_group=id_dict['resourceGroups']))
```

Comment:
when the nic in secondary resource group contains a security group, will there be any bug to find the security group?

Reply:
the logic is 
```
if nic
   attach nic
else
   create_nic_in_vm_rg(nsg in vm rg, 
                                    provided virtual network or vn in vm rg,
                                    provided subnet)
```

That means, if user want to create a nsg refer vn in another resource group, the NIC will attach the NSG found in VM's rg, not the subnet attached one.

Reply:
no creating new nsg, but an exsiting nsg  attached to an existing nic, both of them are in another resource group. then attach the nic to this vm. will this logic cause updating failed to find security group etc?

Reply:
no, the vm only cares the nic's resource id if it successfully find the nic